Add VS Code to Using-IDE page

DanTup opened the pull request:
@mit-mit 

https://dantup-flutter-staging.firebaseapp.com/using-ide/

Currently VS Code content is on another page (to avoid having to manually build the TOC, since the Kramdown plugin doesn't appear to support multiple in one page).

I basically just copied the IntelliJ stuff and adapted it (stripping out bits that weren't relevant, like Inspector).

I also had to tweak the tabs JS to support these "page based" tabs so there are a couple minor tweaks in `jabs.js`.

devoncarew commented on using-ide-vscode.md:
```diff
@@ -0,0 +1,234 @@
+<div class="tabs__content current" markdown="1">
+
+
+The Dart extension provides a fully integrated development experience in the
```
perhaps `in Visual Studio Code.` or `in VS Code.`

devoncarew commented on using-ide-vscode.md:
```diff
@@ -0,0 +1,234 @@
+## Installation and setup
+
+Please follow the [Editor Setup](/get-started/editor/) instructions to install
+the Dart extension.
```
perhaps add `; this extension includes Flutter functionality as well`, or similar

devoncarew commented on using-ide-vscode.md:
```diff
@@ -0,0 +1,234 @@
+1. Enter your desired **Project name**.
+1. Select a **Project location**.
+
+### Creating a new project from existing source code
```
`Opening a project from existing source code`?

devoncarew commented on using-ide-vscode.md:
```diff
@@ -0,0 +1,234 @@
+
+### Creating a new project from existing source code
+
+To open a Flutter project containing existing Flutter source code
```
To open an existing Flutter project

devoncarew commented on using-ide-vscode.md:
```diff
@@ -0,0 +1,234 @@
+* Prefix `stless`: Create a new subclass of `StatelessWidget`.
+* Prefix `stful`: Create a new subclass of `StatefulWidget` and it's associated
+ State subclass.
+* Prefix `stanim`: Create a new subclass of `StatefulWidget`, and it's
```
not for this PR, but I really don't know how common this template is; we should probably remove from VS Code and IntelliJ

/cc @stevemessick, for thoughts on whether we should keep it

DanTup commented:
@devoncarew I've made tweaks based on your comments (some of these were from the IntelliJ doc so I don't know whether any should be tweaked there).

> I have concerns about the duplication with the IntelliJ docs

Yeah, me too. Tbh, this page is pretty long, I don't know if it might be better split up anyway. I'm not sure if people would come to the site looking for a "how to use the IDE" page versus specific things like debugging with breakpoints, creating/opening projects, using snippets, etc.

DanTup commented:
(stage site doesn't have the changes yet - can't do it from corp laptop and the other one has decided to update itself...)

DanTup commented:
I've rebased over current master (to fix Travis from the deprecation errors) and re-staged:

https://dantup-flutter-staging.firebaseapp.com/using-ide-vscode/

DanTup commented:
Doh! That was fixed recently, but there was a PR (which passed at the time it was raised) merged in. I've opened https://github.com/flutter/flutter/issues/16153 and I'll take a look.

goderbauer commented:
https://github.com/flutter/website/pull/922 fixes the build.

DanTup commented:
Rebased again, Travis is green for this PR now (waiting for feedback from @mit-mit, he's been OoO).

mit-mit commented on using-ide-vscode.md:
```diff
@@ -0,0 +1,231 @@
+
+## Installation and setup
+
+Please follow the [Editor Setup](/get-started/editor/) instructions to install
```
Should this link to https://flutter.io/get-started/editor/#vscode ?

DanTup replied:
I did that originally, but the link checker complains because there's no element with the ID `vscode` in that page. I couldn't think of an easy way to fix :(

mit-mit replied:
How do those link ids actually work? Looking at the generated jekyll output in `_site` I don't see `id` tags for them!?

DanTup replied:
We use javascript to read the fragment from the url at page load, and then select the correct tab based on it. If we put real anchors in the page, it would try to scroll down which we don't want. What the link checker is doing seems reasonable, it's just not really valid for this case :(

mit-mit replied:
Got it. Maybe just mark those links to be ignored in the [`Rakefile`](https://github.com/flutter/website/blob/master/Rakefile)?

DanTup replied:
Aha! I was looking around for whether I could configure it. Thanks!

DanTup replied:
Actually looks like you're already skipping these so I've no idea why it was failing for me. I'll do some digging!

DanTup replied:
Ok, seems like the exclusion we had may have been for the exact string "#" so I've changed it to /#/ which allows `#vscode` through, but it makes me think that all other hash fragments were previously being checked and the ignore rules wasn't really needed. Should I leave as being ignored, or see if I can make it more specific (only include #vscode) so it checks more links?

Also, seems like we have a load of 404s linking to developer.apple.com now? :(

mit-mit commented on using-ide-vscode.md:
```diff
@@ -0,0 +1,231 @@
+
+To install updates manually:
+
+1. Click on the Extensions button in the side bar
```
Isn't it 'sidebar' ?

https://en.wikipedia.org/wiki/Sidebar_(computing)

DanTup replied:
Hmm, VS calls it "Side Bar"... WDYT?

<img width="279" alt="screen shot 2018-04-05 at 08 44 23" src="https://user-images.githubusercontent.com/1078012/38353180-9596c944-38ad-11e8-9bc7-97939efd38fd.png">

mit-mit replied:
Oh, yes they do: https://code.visualstudio.com/docs/getstarted/userinterface

But I think we should use caps then -- Side Bar -- to be consistent.

DanTup replied:
Done!

mit-mit commented on using-ide-vscode.md:
```diff
@@ -0,0 +1,231 @@
+
+### Selecting a target device
+
+When a Flutter project is open in VS Code, you should see a set of Flutter
```
Maybe add a screenshot for this, it can be a bit hard to locate.

mit-mit commented on using-ide-vscode.md:
```diff
@@ -0,0 +1,231 @@
+### Run app with breakpoints
+
+1. If desired, set breakpoints in your source code.
+1. CClick **Debug>Start Debugging** in main IDE window or press `F5`.
```
s/CClick/Click

mit-mit commented on using-ide-vscode.md:
```diff
@@ -0,0 +1,231 @@
+
+### Debugging visual layout issues
+
+During a debug session, several debugging commands will be added to the **Command Palette**, including:
```
Maybe screenshot the command palette

DanTup commented:
Ok, think I've addressed all the points, and re-staged here:

https://dantup-flutter-staging.firebaseapp.com/using-ide-vscode/#selecting-a-target-device

@mit-mit not so sure about that last screenshot of the command palette as it has non-debugging things in there (and I didn't want to crop the top off, it'd just be a weird list then. WDYT?

mit-mit commented:
Maybe alternatively just link 'Command Palette' to https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette ?

DanTup commented:
@mit-mit Travis failing on this:

```text
  *  External link https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppDistributionGuide/ConfiguringYourApp/ConfiguringYourApp.html failed: 404 No error
  *  External link https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppDistributionGuide/DistributingYourAppUsingTestFlight/DistributingYourAppUsingTestFlight.html failed: 404 No error
  *  External link https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppDistributionGuide/Introduction/Introduction.html failed: 404 No error
  *  External link https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppDistributionGuide/MaintainingCertificates/MaintainingCertificates.html failed: 404 No error
  *  External link https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppDistributionGuide/SubmittingYourApp/SubmittingYourApp.html failed: 404 No error
  *  External link https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppDistributionGuide/Troubleshooting/Troubleshooting.html failed: 404 No error
  *  External link https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppDistributionGuide/UploadingYourApptoiTunesConnect/UploadingYourApptoiTunesConnect.html failed: 404 No error
  *  External link https://developer.apple.com/library/content/documentation/LanguagesUtilities/Conceptual/iTunesConnect_Guide/Chapters/About.html failed: 404 No error
  *  External link https://developer.apple.com/library/content/documentation/LanguagesUtilities/Conceptual/iTunesConnect_Guide/Chapters/CreatingiTunesConnectRecord.html failed: 404 No error
```

mit-mit commented:
apple 404s look entirely unrelated to your changes; even google search hasn't caught up yet (first link shown here 404s): https://www.google.dk/search?q=%22Distributing+Your+App+Using+TestFlight%22+site%3Aapple.com&oq=%22Distributing+Your+App+Using+TestFlight%22+site%3Aapple.com

mit-mit commented:
I'm going to merge this. Can you please file a bug on the apple links issue? It's also failing in master now: https://travis-ci.org/flutter/website/builds/362327857

DanTup commented:
Great, thanks!

Issue opened at https://github.com/flutter/flutter/issues/16268
